src/main/scala/mesosphere/marathon/core/task/update/impl/steps/ScaleAppUpdateStepImpl.scala

```diff
@@ -45,8 +44,9 @@ class ScaleAppUpdateStepImpl @Inject() (
     terminalOrExpungedTask.foreach { task =>
       val appId = task.taskId.runSpecId
       val taskId = task.taskId
-      val state = task.mesosStatus.fold(TaskState.TASK_STAGING)(_.getState)
-      val reason = task.mesosStatus.fold("")(status => if (status.hasReason) status.getReason.toString else "")
+      val state = task.status.taskStatus.toMesosStateName
```

Comment:
do tests fails if you don't do `toMesosStateName`? Otherwise, for the log message using the taskStatus would be fine